src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,250 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+import jenkins.model.GlobalConfiguration;
+
+import hudson.Extension;
+
+import hudson.model.Build;
+import hudson.model.Result;
+import hudson.model.Project;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.UnprotectedRootAction;
+
+import hudson.security.ACL;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.JsonResponse;
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+
+
+
+@Extension
+public class WebhookEndpoint implements UnprotectedRootAction {
+
+    private String slackUser;
```

Comment:
Just noticed that it looks like WebhookEndpoint is a singleton; there is only ever 1 instantiated per Jenkins instance. If I'm correct, you have a race condition on the slackUser member variable. I think you'll need to pass it around in method calls instead of using a member variable. Maybe you could encapsulate all the required information in an object.

The globalConfig seems fine. I was concerned it would not dynamically pickup configuration changes, but it does.